src/lib/replication/service/raft_repl_service.cpp

```diff
@@ -156,17 +156,20 @@ void RaftReplService::start() {
     LOGINFO("Starting DataService");
     hs()->data_service().start();
 
-    // Step 6: Iterate all the repl dev and ask each one of the join the raft group.
-    for (auto it = m_rd_map.begin(); it != m_rd_map.end();) {
-        auto rdev = std::dynamic_pointer_cast< RaftReplDev >(it->second);
-        rdev->wait_for_logstore_ready();
-        if (!rdev->join_group()) {
-            HS_REL_ASSERT(false, "FAILED TO JOIN GROUP, PANIC HERE");
-            it = m_rd_map.erase(it);
-        } else {
-            ++it;
-        }
-    }
+    // Step 6: Iterate all the repl devs and ask each one of them to join the raft group concurrently.
+    std::vector< std::future< bool > > join_group_futures;
+    for (const auto& [_, repl_dev] : m_rd_map)
```

Comment:
nit: would be nice to put {...} for the `for` loop even thought it is one line. We had a discussion long back that data path code would like to establish this coding convention to avoid making mistakes when more lines needs to be added inside the for loop in the future. 

Similar comment for line: 171

Reply:
ack